system/jlib/jstats.cpp

```diff
@@ -1783,11 +1786,78 @@ class SortedCollectionIterator : public ArrayIIteratorOf<IArrayOf<IStatisticColl
 class CStatisticCollection : public CInterfaceOf<IStatisticCollection>
 {
     friend class CollectionHashTable;
+
+    CStatisticCollection * ensureSubScopePath(std::initializer_list<const StatsScopeId> path)
+    {
+        CStatisticCollection * curScope = this;
+        for (const auto & scopeItem: path)
+            curScope = curScope->ensureSubScope(scopeItem, true); // n.b. this will always return a valid pointer
+        return curScope;
+    }
+    void markDirty()
+    {
+        isDirty=true;
+        if (parent) parent->markDirty();
+    }
+    void refreshAggregates(CRuntimeStatisticCollection & parentTotals, std::function<void(const char * scope, StatisticScopeType sst, StatisticKind kind, stat_type value)> & fWhenAggregateUpdated)
```

Comment:
related to other comments.
I think this can and should be moved outside of CStatisticCollection and into the StatisticAggregator class, and for it to be implemented using IStatisticCollection methods.
As per other comments, IStatisticCollection would still need to expose some methods, but it would be separate out the aggregate implementation, and should be cleaner.